lib/MilkCheck/Engine/BaseEntity.py

```diff
@@ -421,13 +421,13 @@ def remove_dep(self, dep_name, parent=True):
 
     def clear_parent_deps(self):
         '''Remove all parent dependencies of an entity'''
-        for dpname in self.parents.keys():
-            self.remove_dep(dpname)
+        dpnames = self.parents.copy().keys()
```

Comment:
probably easier to read to just add a `list()` call

    for dpname in list(self.parents.keys()):

Reply:
Done